dynawo/sources/Modeler/Common/DYNSubModel.cpp

```diff
@@ -127,6 +127,11 @@ SubModel::setTimeline(const shared_ptr<Timeline>& timeline) {
   timeline_ = timeline;
 }
 
+boost::shared_ptr<timeline::Timeline>
+SubModel::getTimeline() const {
```

Comment:
I would prefer bool hasTimeline(); if this getter is only used to test if pointer is NULL, @rosiereflo what is your take on this ?

Reply:
I agree

Reply:
I agree too, but getTimeline() is used in new version if DYNAddEvent(), the macro, to gain access to the timeline :
DYNMacrosMessage.h:            timeline::Timeline& tl(*(pNetwork)->getTimeline());

Reply:
Hi, Florent has taken over from your original branch and simplified the macros, so I think that they do not use this method anymore. If I'm not mistaken with the current version Gautier is right we can replace this method with a hasTimeline

Reply:
correct

Reply:
Thank you Florent :) I don't follow anymore. Did you see I added an unitary test for the macro? See you